integration/setup/setup.go

```diff
@@ -43,7 +44,26 @@ func setup(ctx context.Context, m *testing.M, config Config) (int, error) {
 			defer func() {
 				err := config.K8sSetup.EnsureNamespaceDeleted(ctx, key.Namespace)
 				if err != nil {
-					config.Logger.LogCtx(ctx, "level", "error", "message", fmt.Sprintf("failed to delete namespace %#q", key.Namespace), "stack", fmt.Sprintf("%#v", err))
+					config.Logger.LogCtx(ctx, "level", "error", "message", fmt.Sprintf("failed to delete Namespace %#q", key.Namespace), "stack", fmt.Sprintf("%#v", err))
+				}
+			}()
+		}
+	}
+
+	// Create App CRD
+	{
+		crd := applicationv1alpha1.NewAppCRD()
+
+		err := config.K8sSetup.EnsureCRDCreated(ctx, crd)
+		if err != nil {
+			return 0, microerror.Mask(err)
+		}
+
+		if !env.CircleCI() && !env.KeepResources() {
```

Comment:
Why do we need to check if we're in circle? Because the whole thing gets torn down then anyway?

Reply:
Yes.